ocaml/xapi/xapi_host.ml

```diff
@@ -2683,24 +2683,59 @@ let write_uefi_certificates_to_disk ~__context ~host:_ =
     | Ok contents ->
         (* Remove existing certs before extracting xapi ones
          * to avoid a extract override issue. *)
+        let auth_files = ["PK.auth"; "KEK.auth"; "db.auth"; "dbx.auth"] in
         List.iter
           (fun name ->
             let path = Filename.concat !Xapi_globs.varstore_dir name in
             Unixext.unlink_safe path
           )
-          ["KEK.auth"; "db.auth"] ;
+          auth_files ;
         (* No uefi certificates, nothing to do. *)
         if contents <> "" then (
           with_temp_file_contents ~contents
             (Tar_unix.Archive.extract extract_certificate_file) ;
           debug "UEFI tar file extracted to temporary directory"
-        )
+        ) ;
+        (* If all auth files are missing, fetch them from a default dir *)
+        if
```

Comment:
I tried to explain the use case here: https://github.com/xapi-project/xen-api/issues/4822

Users who take no particular action will always get the default auth files. Users who imports certs to XAPI are supposed to be knowing what they are doing, so we give them full control. If some auths are present and others are missing, it could be on purpose (for example "I don't want a dbx"), and even if it's a mistake, it would be confusing if, say, a missing db is replaced with that from `/usr/share/varstored`, if it's not been signed with the KEK that was imported by the user.

We also provide a helper script to users which lets them import certificates from CLI and enforces the presence of at least PK, KEK and db for the common case, so the risk of error is low in our context.

Reply:
> Only when all auth files are missing, they are copied from a back up. How can we arrive in this situation in the first place?

Now that `varstored` will embed the default certs, an empty `pool.uefi_certificates` will be the common case. No more action will be required from users to use Guest Secure Boot (no more helper that harvests the certs from the firmware and sets them in XAPI in XenSever, no more requirement to run `secureboot-certs install` to download the certs on XCP-ng.

Only users with specific needs will call `pool.set_uefi_certificates`, and in XCP-ng's case they won't even need to do it directly, as we provide a `secureboot-certs` helper to pack (with create-auth) and install the custom certificates.